deployment/terraform/cloudfront-functions/conditional_oshub_redirect.js

```diff
@@ -0,0 +1,71 @@
+// Adapted from https://devscope.io/code/aws-samples/amazon-cloudfront-functions/issues/11
+
+function objectToQueryString(obj) {
+    var str = [];
+    for (var param in obj)
+        if (obj[param].value === '') {
+            str.push(encodeURIComponent(param));
+        } else {
+            // Normalize single and multiple value query params to a list of
+            // param value objects
+            var valueObjects = obj[param].multiValue || [obj[param]];
+            var encodedValues = valueObjects.map(
+                vo => `${encodeURIComponent(param)}=${encodeURIComponent(vo.value)}`);
+
+            str.push(encodedValues.join('&'));
+        }
+
+    return str.join('&');
+}
+
+function handler(event) {
+    var request = event.request;
+    var uri = request.uri;
+
+    // If this is an embed request, allow it to proceed to the ALB
+    if (request.querystring.embed && request.querystring.embed.value === '1') {
+        return request;
+    }
+
+
+    // Ensure that any resources fetched by the client after when loading an
+    // ebedded map are not redirected
+    if (uri.substring(1, 5) === 'web/'
+        || uri.substring(1, 5) === 'api/'
+        || uri.substring(1, 19) === 'api-feature-flags/'
+        || uri.substring(1, 10) === 'facvicon/'
+        || uri.substring(1, 8) === 'images/'
+        || uri.substring(1, 8) === 'static/'
+       ) {
+        // If the referrer starts with one of the following then we want to
+        // allow the request to pass through unmodified
```

Comment:
Why do we let these two pass through?

Reply:
This comment refers to the two comment lines below

```
        //     https://staging.openapparel.org/
        //     https://openapparel.org/
```

When the embedded map page successfully loaded from either of these two domains makes a request to /api/* to load data for dropdowns, etc. the referer header is set, so we let the request pass through.